core/src/main/java/org/vertexium/FetchHints.java

```diff
@@ -414,4 +417,103 @@ private boolean isEdgeLabelsOfEdgeRefsIncluded(ImmutableSet<String> edgeLabelsOf
         }
         return true;
     }
+
+    public static FetchHints union(FetchHints... fetchHints) {
+        return union(Arrays.asList(fetchHints));
+    }
+
+    public static FetchHints union(Iterable<FetchHints> fetchHints) {
+        boolean includeAllProperties = false;
+        Set<String> propertyNamesToInclude = null;
+        boolean includeAllPropertyMetadata = false;
+        Set<String> metadataKeysToInclude = null;
+        Boolean includeHidden = null;
+        boolean includeAllEdgeRefs = false;
+        boolean includeOutEdgeRefs = false;
+        boolean includeInEdgeRefs = false;
+        boolean includeEdgeLabelsAndCounts = false;
+        boolean includeExtendedDataTableNames = false;
+        Boolean ignoreAdditionalVisibilities = null;
+        Boolean includePreviousMetadata = null;
+        Set<String> edgeLabelsOfEdgeRefsToInclude = null;
+
+        for (FetchHints fetchHint : fetchHints) {
+            if (fetchHint.isIncludeAllProperties()) {
+                includeAllProperties = true;
+            }
+            if (fetchHint.isIncludeAllPropertyMetadata()) {
+                includeAllPropertyMetadata = true;
+            }
+            if (fetchHint.isIncludeAllEdgeRefs()) {
+                includeAllEdgeRefs = true;
+            }
+            if (fetchHint.isIncludeOutEdgeRefs()) {
+                includeOutEdgeRefs = true;
+            }
+            if (fetchHint.isIncludeInEdgeRefs()) {
+                includeInEdgeRefs = true;
+            }
+            if (fetchHint.isIncludeEdgeLabelsAndCounts()) {
+                includeEdgeLabelsAndCounts = true;
+            }
+            if (fetchHint.isIncludeExtendedDataTableNames()) {
+                includeExtendedDataTableNames = true;
+            }
+
+            if (includeHidden != null && includeHidden != fetchHint.isIncludeHidden()) {
+                throw new VertexiumException("Incompatible fetch hints to combine (includeHidden).");
+            }
+            includeHidden = fetchHint.isIncludeHidden();
+
+            if (ignoreAdditionalVisibilities != null && ignoreAdditionalVisibilities != fetchHint.isIgnoreAdditionalVisibilities()) {
+                throw new VertexiumException("Incompatible fetch hints to combine (ignoreAdditionalVisibilities).");
+            }
+            ignoreAdditionalVisibilities = fetchHint.isIgnoreAdditionalVisibilities();
+
+            if (includePreviousMetadata != null && includePreviousMetadata != fetchHint.isIncludePreviousMetadata()) {
+                throw new VertexiumException("Incompatible fetch hints to combine (includePreviousMetadata).");
+            }
+            includePreviousMetadata = fetchHint.isIncludePreviousMetadata();
+
+            if (fetchHint.getPropertyNamesToInclude() != null) {
```

Comment:
Should we also skip this if `isIncludeAllProperties` is true? If the answer to that is yes, should we clean out `propertyNamesToInclude` if we find a true `isIncludeAllProperties` mid-way through the iteration?

Reply:
I ended up putting this logic in the constructor since it should really work with the builder as well. https://github.com/visallo/vertexium/pull/418/files#diff-804687a9c91cd00ca960efeb91ac898aR73